incomplete_cooperative/bounds.py

```diff
@@ -71,6 +69,31 @@ def compute_bounds_superadditive_cached(game: BoundableIncompleteGame) -> None:
         game.set_upper_bound(upper_bound, Coalition(coalition))
 
 
+def compute_bounds_superadditive_monotone_approx_cached(game: BoundableIncompleteGame, repetitions: int) -> None:
+    """Compute the bounds given a superadditive incomplete game."""
+    assert game.is_value_known(Coalition(0))
+    assert game.is_value_known(Coalition(2**game.number_of_players - 1))
+    all_coalitions, all_sorted, coal_structure = _get_sub_super_coalition_structure(game.number_of_players)
+    unknown_sorted = all_sorted[np.logical_not(game.are_values_known()[all_sorted])]
+    for _ in range(repetitions):
+        for coalition in unknown_sorted:
+            super_coalitions = all_coalitions[coal_structure[coalition] == 1]
```

Comment:
The `_get_sub_super_coalition_structure` Gives us an array where `coal_structure[coalition, other]` is `1` if `other` is a supercoalition of `coalition`, 2 if it is a subcoalition, 0 if it is equal, and -1 otherwise.